core-api/src/main/java/com/optimizely/ab/config/parser/ConditionJacksonDeserializer.java

```diff
@@ -81,8 +98,8 @@ else if (subNode.isObject()) {
             case "or":
                 condition = new OrCondition(conditions);
                 break;
-            case "not": // this makes two assumptions: operator is "not" and conditions is non-empty...
-                condition = new NotCondition(conditions.get(0));
+            case "not":
+                condition = new NotCondition(conditions.isEmpty() ? new EmptyCondition() : conditions.get(0));
```

Comment:
`EmptyCondition` evaluates to true, which isn't quite what we want: a `"not"` with no operands is invalid and should evaluate to null.

Perhaps we could define a new `InvalidCondition` that evaluates to true, or construct `new NotCondition(conditions.get(0))` as we were doing before while also updating `NotCondition` to correctly handle a nullable arg.

(Applies to ConditionUtils.java too)